[JavaScript] Add array-transformations Concept Exercise (#1951)
exercism/v3: languages/javascript/exercises/concept/array-transformations/.docs/instructions.md

> Elyse, magician-to-be, continues her training. She will be given a deck of cards and attempt to transform that deck. She will make certain cards appear, disappear, change their values, or completely rearrange themselves.

Reviewer: I like how the existing theme is being re-used!

Author: Just following the instructions. :)

To make things easier, she usually only starts with cards numbered 1 to 10, although some of the tricks may result in larger cards being created.

> ## 1. - Double every single card

Reviewer: ```suggestion
## 1. Double every single card
```

Elyse wants to double the number of every card in the deck. This may result in higher cards than 1-10.

```javascript
const deck = [1, 2, 3, 4, 10]

> seeingDouble(deck)

Reviewer: I like the naming! :D

Author: Thanks, I tried to make it light and fun.

// => [2, 4, 6, 8, 20]
```

> ## 2. - Create multiple copies of every 3 found in the deck

Reviewer: ```suggestion
## 2. Create multiple copies of every 3 found in the deck
```

> Elyse wants to triplicate ever 3 found in the deck. If a deck started with a single 3, after the trick the deck would have three 3's in place of original.

Reviewer 2: ```suggestion
Elyse wants to triplicate every 3 found in the deck. If a deck started with a single 3, after the trick the deck would have three 3's in place of original.
```

```javascript
const deck = [1, 3, 9, 3, 7]
threeOfEachThree(deck)
// => [1, 3, 3, 3, 9, 3, 3, 3, 7]
```

> ## 3. - Find two cards from the exact middle of the deck

Reviewer: ```suggestion
## 3. Find two cards from the exact middle of the deck
```

> Elyse will take a deck and make every card disappear except the middle two cards. This trick is always performed with exactly 10 cards.

Reviewer 2: ```suggestion
Elyse will take a deck of ten cards, and make every card disappear except the middle two cards. 
```

```javascript
const deck = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
middleTwo(deck)
// => [5, 6]
```

> ## 4. - The outside two cards will reappear in the middle of the deck

Reviewer: ```suggestion
## 4. The outside two cards will reappear in the middle of the deck
```

Elyse wants to move the top and bottom cards of the deck into the middle, in reverse order.

```javascript
const deck = [1, 2, 3, 5, 6, 10]
sandwichTrick(deck)
// => [2, 3, 10, 1, 5, 6]
```

> ## 5. - Every card that isn't 2 disappears

Reviewer: ```suggestion
## 5. Every card that isn't 2 disappears
```

> Elyse's favorite number today is 2. In this trick every card that isn't a 2 will disappear from the eck.

Reviewer 2: ```suggestion
Elyse's favorite number today is 2. In this trick every card that isn't a 2 will disappear from the deck.
```

```javascript
const deck = [1, 2, 3, 4, 10, 2]
twoIsSpecial(deck)
// => [2, 2]
```

> ## 6. - Convert a shuffled deck into a perfectly ordered deck

Reviewer: ```suggestion
## 6. Convert a shuffled deck into a perfectly ordered deck
```
